packages/lesswrong/server/migrations/2022-08-31-recomputeWordCounts.ts

```diff
@@ -0,0 +1,76 @@
+import { registerMigration, forEachDocumentBatchInCollection } from "./migrationUtils";
+import { editableCollections, editableCollectionsFields } from "../../lib/editor/make_editable"
+import { getCollection } from "../../lib/vulcan-lib";
+import { dataToWordCount } from "../editor/make_editable_callbacks";
+import { Revisions } from "../../lib/collections/revisions/collection";
+
+registerMigration({
+  name: "recomputeWordCounts",
+  dateWritten: "2022-08-31",
+  idempotent: true,
+  action: async () => {
+    await forEachDocumentBatchInCollection({
+      collection: Revisions,
+      batchSize: 1000,
+      callback: async (documents: DbRevision[]) => {
+        const updates: Array<any> = [];
+        for (const doc of documents) {
+          const { data, type } = doc.originalContents;
+          const wordCount = await dataToWordCount(data, type);
+          if (wordCount !== doc.wordCount) {
+            updates.push({
+              updateOne: {
+                filter: { _id: doc._id },
+                update: {
+                  $set: {
+                    wordCount: wordCount
+                  }
+                }
+              }
+            });
+          }
+        }
+        // eslint-disable-next-line no-console
+        console.log(`Writing ${updates.length} word counts for Revisions`);
+        if (updates.length) {
+          await Revisions.rawCollection().bulkWrite(updates, { ordered: false });
+        }
+      },
+    });
+    for (const collectionName of editableCollections) {
```

Comment:
You modeled this off of `2019-02-14-computeWordCounts.ts` right? Could you add more comments in this file explaining what this is doing?

Also, maybe we could just call this on the post contents field, instead of all editable fields? Is that the only field we really care about?

Reply:
I just added a brief comment to the file describing how it works and why it's separate from the other migration - let me know if you think it needs any more detail.

I think you're right that we only need to update revisions and post bodies in this case. I'll leave the migration as is though because it may be a useful utility in the future, and I'll just filter out the other collections before I run it.